service/src/main/kotlin/no/nav/su/se/bakover/service/søknad/SøknadServiceImpl.kt

```diff
@@ -120,7 +120,24 @@ internal class SøknadServiceImpl(
         )
     }
 
-    override fun hentSøknad(søknadId: UUID): Either<KunneIkkeLukkeSøknad.FantIkkeSøknad, Søknad> {
-        return søknadRepo.hentSøknad(søknadId)?.right() ?: KunneIkkeLukkeSøknad.FantIkkeSøknad.left()
+    override fun hentSøknad(søknadId: UUID): Either<FantIkkeSøknad, Søknad> {
+        return søknadRepo.hentSøknad(søknadId)?.right() ?: FantIkkeSøknad.left()
+    }
+
+    override fun lagUtskrift(søknadId: UUID): Either<KunneIkkeLageSøknadsutskrift, ByteArray> {
+        val søknad = hentSøknad(søknadId).getOrElse {
+            log.error("Skriv ut søknad: Fant ikke søknad")
+            return KunneIkkeLageSøknadsutskrift.FantIkkeSøknad.left()
+        }
+
+        return pdfGenerator.genererPdf(søknad.søknadInnhold).fold(
+            {
+                log.error("Skriv ut søknad: Kunne ikke generere PDF. Originalfeil: $it")
+                KunneIkkeLageSøknadsutskrift.KunneIkkeLagePdf.left()
+            },
+            {
+                it.right()
+            }
+        )
```

Comment:
Dette kan skrives litt smoothere:
```suggestion
        return hentSøknad(søknadId)
            .mapLeft {
                log.error("Skriv ut søknad: Fant ikke søknad")
                KunneIkkeLageSøknadsutskrift.FantIkkeSøknad
            }
            .flatMap { søknad ->
              pdfGenerator.genererPdf(søknad.søknadInnhold)
                  .mapLeft {
                      log.error("Skriv ut søknad: Kunne ikke generere PDF. Originalfeil: $it")
                          KunneIkkeLageSøknadsutskrift.KunneIkkeLagePdf
                  }
            }
```

Reply:
Ops, suggestionen blev lite outdated när jag gjorde endringer. Men kan prøva apply:a den manuellt 😎